code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADConfigManager.java

```diff
@@ -145,40 +163,77 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
+        settingsUtils.changeSetting(ADConstants.DOMAIN_SETTING, config.get(DOMAIN));
+        settingsUtils.changeSetting(ADConstants.GROUP_NAME_FIELD_SETTING, config.get("groupNameField"));
+        settingsUtils.changeSetting(ADConstants.GROUP_OBJECT_CLASS_SETTING, config.get("groupObjectClass"));
+        settingsUtils.changeSetting(ADConstants.GROUP_SEARCH_FIELD_SETTING, config.get("groupSearchField"));
+        settingsUtils.changeSetting(ADConstants.LOGIN_DOMAIN_SETTING, config.get("loginDomain"));
+        settingsUtils.changeSetting(ADConstants.PORT_SETTING, config.get("port"));
+        settingsUtils.changeSetting(ADConstants.SERVER_SETTING, config.get("server"));
+        if(config.get("serviceAccountPassword") != null){
+            settingsUtils.changeSetting(ADConstants.SERVICE_ACCOUNT_PASSWORD_SETTING, config.get("serviceAccountPassword"));
+        }
+        settingsUtils.changeSetting(ADConstants.SERVICE_ACCOUNT_USERNAME_SETTING, config.get("serviceAccountUsername"));
+        settingsUtils.changeSetting(ADConstants.TLS_SETTING, config.get("tls"));
+        settingsUtils.changeSetting(ADConstants.USER_DISABLED_BIT_MASK_SETTING, config.get("userDisabledBitMask"));
+        settingsUtils.changeSetting(ADConstants.USER_ENABLED_ATTRIBUTE_SETTING, config.get("userEnabledAttribute"));
+        settingsUtils.changeSetting(ADConstants.USER_LOGIN_FIELD_SETTING, config.get("userLoginField"));
+        settingsUtils.changeSetting(ADConstants.USER_NAME_FIELD_SETTING, config.get("userNameField"));
+        settingsUtils.changeSetting(ADConstants.USER_OBJECT_CLASS_SETTING, config.get("userObjectClass"));
+        settingsUtils.changeSetting(ADConstants.USER_SEARCH_FIELD_SETTING, config.get("userSearchField"));
+        settingsUtils.changeSetting(ADConstants.TIMEOUT_SETTING, config.get("connectionTimeout"));
+        settingsUtils.changeSetting(SecurityConstants.SECURITY_SETTING, config.get("enabled"));
+
+        if (config.get("enabled") != null){
             settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, ADConstants.CONFIG);
         } else {
             settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, SecurityConstants.NO_PROVIDER);
         }
+
+        String ids = validateIdentities((List<Map<String, String>>) config.get(ALLOWED_IDENTITIES));
+        settingsUtils.changeSetting(ADConstants.ALLOWED_IDENTITIES_SETTING, ids);
+
         return currentLdapConfig(config);
     }
 
+    public String validateIdentities(List<Map<String, String>> identitiesGiven) {
+        StringBuilder sb = new StringBuilder();
+        List<Identity> identities = getIdentities(identitiesGiven);
+        Iterator<Identity> identityIterator = identities.iterator();
+        while (identityIterator.hasNext()){
+            sb.append(identityIterator.next().getId().trim());
+            if (identityIterator.hasNext()) sb.append('#');
```

Comment:
yes we need some character to separate the ldap ids and save it to db in a string. We cannot use a "," or ";" character as a separator of ldap Ids - which are CNs(common names) of the format CN=YourName,OU=Users,DC=YourDomain,DC=com - already separated by "," . And a ";" may also be used in a CN.
So am using a "#" as a separator for now.